diff --git a/Tables/dbo.Invoice.sql b/Tables/dbo.Invoice.sql
index 19a8243..e7cdcfb 100644
--- a/Tables/dbo.Invoice.sql
+++ b/Tables/dbo.Invoice.sql
@@ -4,8 +4,6 @@
 [InvoiceNumber] [varchar] (100) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
 [InvoiceDate] [date] NOT NULL,
 [Amount] [money] NOT NULL,
-[PayorID] [int] NOT NULL,
-[ClaimID] [int] NOT NULL,
 [CreatedOnUTC] [datetime2] NOT NULL CONSTRAINT [dfInvoiceCreatedOnUTC] DEFAULT (sysutcdatetime()),
 [UpdatedOnUTC] [datetime2] NOT NULL CONSTRAINT [dfInvoiceUpdatedOnUTC] DEFAULT (sysutcdatetime()),
 [DataVersion] [timestamp] NOT NULL,
@@ -17,10 +15,4 @@
 )
 GO
 ALTER TABLE [dbo].[Invoice] ADD CONSTRAINT [pkInvoice] PRIMARY KEY CLUSTERED  ([InvoiceID]) WITH (FILLFACTOR=90, DATA_COMPRESSION = ROW) ON [PRIMARY]
-GO
-CREATE NONCLUSTERED INDEX [idxInvoiceClaimIDClaimClaimIDIncludeAll] ON [dbo].[Invoice] ([ClaimID], [PayorID]) INCLUDE ([Amount], [CreatedOnUTC], [InvoiceDate], [InvoiceID], [InvoiceNumber], [UpdatedOnUTC]) WITH (DATA_COMPRESSION = PAGE) ON [PRIMARY]
-GO
-ALTER TABLE [dbo].[Invoice] ADD CONSTRAINT [fkInvoiceClaimIDClaimClaimID] FOREIGN KEY ([ClaimID]) REFERENCES [dbo].[Claim] ([ClaimID])
-GO
-ALTER TABLE [dbo].[Invoice] ADD CONSTRAINT [fkInvoicePayorIDPayorPayorID] FOREIGN KEY ([PayorID]) REFERENCES [dbo].[Payor] ([PayorID])
 GO
diff --git a/Tables/dbo.Payment.sql b/Tables/dbo.Payment.sql
deleted file mode 100644
index 520b6bb..0000000
--- a/Tables/dbo.Payment.sql
+++ /dev/null
@@ -1,23 +0,0 @@
-CREATE TABLE [dbo].[Payment]
-(
-[PaymentID] [int] NOT NULL IDENTITY(1, 1),
-[CheckNumber] [varchar] (50) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
-[AmountPaid] [money] NOT NULL,
-[DateScanned] [date] NULL,
-[PrescriptionID] [int] NULL,
-[ClaimID] [int] NOT NULL,
-[CreatedOnUTC] [datetime2] NOT NULL CONSTRAINT [dfPaymentCreatedOnUTC] DEFAULT (sysutcdatetime()),
-[UpdatedOnUTC] [datetime2] NOT NULL CONSTRAINT [dfPaymentUpdatedOnUTC] DEFAULT (sysutcdatetime()),
-[DataVersion] [timestamp] NOT NULL
-) ON [PRIMARY]
-WITH
-(
-DATA_COMPRESSION = ROW
-)
-GO
-ALTER TABLE [dbo].[Payment] ADD CONSTRAINT [pkPayment] PRIMARY KEY CLUSTERED  ([PaymentID]) WITH (FILLFACTOR=90, DATA_COMPRESSION = ROW) ON [PRIMARY]
-GO
-ALTER TABLE [dbo].[Payment] ADD CONSTRAINT [fkPaymentClaimIDClaimClaimID] FOREIGN KEY ([ClaimID]) REFERENCES [dbo].[Claim] ([ClaimID])
-GO
-ALTER TABLE [dbo].[Payment] ADD CONSTRAINT [fkPaymentPrescriptionIDPrescriptionPrescriptionID] FOREIGN KEY ([PrescriptionID]) REFERENCES [dbo].[Prescription] ([PrescriptionID])
-GO
diff --git a/Tables/dbo.Prescription.sql b/Tables/dbo.Prescription.sql
index cc4f577..c44f18d 100644
--- a/Tables/dbo.Prescription.sql
+++ b/Tables/dbo.Prescription.sql
@@ -58,3 +58,5 @@
 GO
 ALTER TABLE [dbo].[Prescription] ADD CONSTRAINT [fkPrescriptionInvoiceIDInvoiceInvoiceID] FOREIGN KEY ([InvoiceID]) REFERENCES [dbo].[Invoice] ([InvoiceID])
 GO
+ALTER TABLE [dbo].[Prescription] ADD CONSTRAINT [fkPrescriptionPharmacyNABPPharmacyNABP] FOREIGN KEY ([PharmacyNABP]) REFERENCES [dbo].[Pharmacy] ([NABP])
+GO
